server/src/main/java/io/spine/server/entity/AbstractEntity.java

```diff
@@ -35,19 +43,26 @@
 import java.util.function.Function;
 
 import static com.google.common.base.Preconditions.checkNotNull;
+import static io.spine.util.Exceptions.newIllegalArgumentException;
 
 /**
  * Abstract base for entities.
  *
  * @param <I> the type of entity identifiers
  * @param <S> the type of entity state objects
- * @author Alexander Yevsyukov
- * @author Dmitry Ganzha
  */
 @SuppressWarnings("SynchronizeOnThis") /* This class uses double-check idiom for lazy init of some
     fields. See Effective Java 2nd Ed. Item #71. */
 public abstract class AbstractEntity<I, S extends Message> implements Entity<I, S> {
 
+    /**
+     * Indicates if the lifecycle flags of the entity were changed since initialization.
+     *
+     * <p>Changed lifecycle flags are should be updated when
+     * {@linkplain io.spine.server.entity.Repository#store(io.spine.server.entity.Entity) storing}.
+     */
+    private volatile boolean lifecycleFlagsChanged;
```

Comment:
Maybe, however, even if we accept this change, we shouldn't implement it in this PR.

Reply:
This is how it was in the entity hierarchy that this PR is merging. Let's discuss the transaction matter separately because this would be a bigger change.